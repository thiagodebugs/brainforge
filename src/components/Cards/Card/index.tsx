import Icon from "@mdi/react";
import Link from "next/link";
import style from "./style.module.css";
import * as mdi from "@mdi/js";

interface CardProps {
  card: {
    title?: string;
    icon: string | keyof typeof mdi;
    description?: string;
    link: string;
  };
}

export default function Card({ card }: CardProps) {
  const iconPath =
    typeof card.icon === "string"
      ? mdi[card.icon as keyof typeof mdi]
      : card.icon;

  return (
    <>
      <Link href={card.link} className={style.card}>
        <Icon path={iconPath} size={1.5} />
        <h3>{card.title}</h3>
        <p>{card.description}</p>
      </Link>
    </>
  );
}
