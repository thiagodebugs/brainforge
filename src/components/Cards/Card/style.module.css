.card {
  @apply flex flex-col bg-neutral-800 rounded-xl p-4 transition-all duration-300 transform lg:hover:scale-105 hover:bg-[#26c666] ease-in-out min-w-36 max-w-sm w-full;
}

.card svg {
  @apply transition-all ease-in-out duration-300 transform;
  color: #26c666;
}

.card:hover svg {
  @apply text-white;
}

.card h3 {
  @apply text-xl font-bold flex-1 mt-4 mb-2;
}

.card p {
  @apply text-xs font-extralight;
}
