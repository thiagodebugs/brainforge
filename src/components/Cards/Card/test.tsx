import { render } from "@testing-library/react";
import Card from ".";

const props = {
  title: "Lorem ipsum",
  icon: "mdiCard",
  description: "Test",
  link: "/test",
};

describe("Card", () => {
  it("should render the card", () => {
    const { container } = render(<Card card={props} />);

    expect(container).toMatchSnapshot();
  });

  it("click on card", () => {
    const { getByText } = render(<Card card={props} />);

    expect(getByText("Lorem ipsum")).toBeInTheDocument();
    expect(getByText("Test")).toBeInTheDocument();
  });
});
