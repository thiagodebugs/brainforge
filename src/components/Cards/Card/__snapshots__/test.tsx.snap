// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render the card 1`] = `
<div>
  <a
    class="card"
    href="/test"
  >
    <svg
      role="presentation"
      style="width: 2.25rem; height: 2.25rem;"
      viewBox="0 0 24 24"
    >
      <path
        d="M20,20H4A2,2 0 0,1 2,18V6A2,2 0 0,1 4,4H20A2,2 0 0,1 22,6V18A2,2 0 0,1 20,20Z"
        style="fill: currentColor;"
      />
    </svg>
    <h3>
      Lorem ipsum
    </h3>
    <p>
      Test
    </p>
  </a>
</div>
`;
